techmatters/flex-plugins
# CHI 1121 Accessibility Enhancements for Case List

Primary reviewer: @[GPaoloni](https://github.com/GPaoloni)

## Description
- Removed 'expand' button and added a new styled component CLCaseIDButton based on button element for CaseId - keyboard buttons, tab and enter/space work
- Styling for CaseId column has been modified according to this [mock](https://marvelapp.com/prototype/6gcabah/screen/86290860)
- Tested that table content is accessible using screen readers.
- Added outline on case rows when hovered on mouse and made clickable anywhere on row
- Rearranged and adjusted the table column widths to accommodate the new mocks. 
### Check Original [PR](https://github.com/techmatters/flex-plugins/pull/732) for these additional fixes:
- Case List Headers are accessible by keyboard and screen readers.
- Sort Arrow Button changed
- Added string parsing for case statuses
- Pending: focus on first case list element

### Checklist
- [n/a] New tests added
- [n/a] Tested for chat contacts
- [n/a] Tested for call contacts

### Related Issues
Fixes [#1121](https://bugs.benetech.org/browse/CHI-1121)

### Verification steps
- Use keyboard to navigate through case list page with tab and open a case with enter or space
- Use Screen Reader  to navigate through case list page and check that detailed information is also accessible
- Check with the updated [mocks](https://marvelapp.com/prototype/6gcabah/screen/86290860) to ensure implementation is identical.
